fix: add missing values in error messages in `blas/base/sgemv`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

none

## Description

> What is the purpose of this pull request?

This pull request:

-   This adds missing interpolated argument values in error messages.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-  none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
test: add test cases for blas/base/sgemv

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/blas/base/sgemv/lib/ndarray.js:
```diff
@@ -72,11 +72,17 @@ function sgemv( trans, M, N, alpha, A, strideA1, strideA2, offsetA, x, strideX,
+	}
 	if ( strideX === 0 ) {
-		throw new RangeError( format( 'invalid argument. Eleventh argument must be non-zero.' ) );
+		throw new RangeError( format( 'invalid argument. Tenth argument must be non-zero.', strideX ) );
```
Thanks for catching!

kgryte commented on lib/node_modules/@stdlib/blas/base/sgemv/lib/ndarray.js:
```diff
@@ -72,11 +72,17 @@ function sgemv( trans, M, N, alpha, A, strideA1, strideA2, offsetA, x, strideX,
 	if ( N < 0 ) {
 		throw new RangeError( format( 'invalid argument. Fourth argument must be a nonnegative integer. Value: `%d`.', N ) );
 	}
+	if ( strideA1 === 0 ) {
```
@ShabiShett07 In contrast to `dtrsv`, we don't have an `LDA` check for `sgemv`. So why are we adding explicit `strideA*` checks here? I don't think that these checks are needed or desired.

kgryte commented on lib/node_modules/@stdlib/blas/base/sgemv/test/test.ndarray.js:
```diff
@@ -145,6 +145,52 @@ tape( 'the function throws an error if provided an invalid third argument', func
 	}
 });
 
+tape( 'the function throws an error if provided an invalid sixth argument', function test( t ) {
```
Same comment. I don't think we need to provide `0` strides for `A`.

kgryte reviewed (changes requested):
Left initial comments. As I mentioned elsewhere, we should not blindly add `strideA*` checks. This needs to be reasoned about on a per routine basis. For this and `dgemv`, I don't think the checks are warranted and they should be removed.

The fixed error messages, however, can stay.

ShabiShett07 commented:
@kgryte, now I got it the implementation needs to have LDA check for having stride checks. Now I am clear with it. Thank you!

ShabiShett07 committed:
fix: update test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, ready for review

kgryte commented on lib/node_modules/@stdlib/blas/base/sgemv/lib/ndarray.js:
```diff
@@ -73,10 +73,10 @@ function sgemv( trans, M, N, alpha, A, strideA1, strideA2, offsetA, x, strideX,
 	}
 	if ( strideX === 0 ) {
-		throw new RangeError( format( 'invalid argument. Eleventh argument must be non-zero.' ) );
+		throw new RangeError( format( 'invalid argument. Tenth argument must be non-zero.', strideX ) );
```
```suggestion
		throw new RangeError( format( 'invalid argument. Tenth argument must be non-zero. Value: `%d`.', strideX ) );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/sgemv/lib/ndarray.js:
```diff
@@ -73,10 +73,10 @@ function sgemv( trans, M, N, alpha, A, strideA1, strideA2, offsetA, x, strideX,
 	}
 	if ( strideY === 0 ) {
-		throw new RangeError( format( 'invalid argument. Fifteenth argument must be non-zero.' ) );
+		throw new RangeError( format( 'invalid argument. Fourteenth argument must be non-zero.', strideY ) );
```
```suggestion
		throw new RangeError( format( 'invalid argument. Fourteenth argument must be non-zero. Value: `%d`.', strideY ) );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>
